src/loaders/tiff/index.ts

```diff
@@ -64,5 +52,5 @@ export async function loadOmeTiff(
    */
   checkProxies(tiff);
 
-  return load(tiff);
+  return pool ? load(tiff, new Pool()) : load(tiff);
```

Comment:
Maybe we should make this a flag instead of an object passed around? That way, the default is just to have a pool (instead of not).  Maybe there's a reason not do this that I am missing/forgetting.

Reply:
I'm a little confused. `pool` is a flag here, and the default is `true` and initializes a new `Pool`.

Reply:
Should have been clearer, my apologies - the flag would continue to be passed down into the constructor for the `TiffPixelSource` so no `Pool` objects are created until initialization.  I guess the downside here would be that each `TiffPixelSource` would have its own `Pool` but I don't think that's particularly problematic?  Maybe it is though.  Maybe we could just wait until the `TiffPixelSource` is made to make the `Pool` object then?  Might just be overthinking this.

Reply:
No worries. Two initial thoughts:

1.) This will make the `TiffPixelSource` a much heaver object than `ZarrPixelSource`. The `PixelSource`s are primarily a small amount of glue code around `geotiff` and `zarr` APIs/Objects. Putting `Pool` in the constructor would mean it's not really an "option" in your bundle (it cannot be statically tree-shaken, along with the large in-lined worker code).

2.) There is less flexibility of how to use the pool. Right now, you can create multiple `GeoTIFF`s and reuse a single `Pool` (ideal); the suggested change would _require_ that a `Pool` is instantiated per pyramidal resolution per `GeoTIFF`.

Reply:
I am fine leaving this, you are right.